### ISSUE TITLE
Doesn't build

### ISSUE THREAD

**Issue by AppreciateIT** [jamesshore/object_playground#7]
On Mac OSX, node version 9.4.0, when I do 

./jake.sh karma

build fails with:

TypeError: Cannot read property 'prototype' of undefined
    at Object.<anonymous> (/PhpstormProjects/ObjectPlayground/object_playground/node_modules/karma/node_modules/socket.io/lib/store.js:35:42)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/PhpstormProjects/ObjectPlayground/object_playground/node_modules/karma/node_modules/socket.io/lib/manager.js:16:13)
    at Module._compile (module.js:660:30)

Any help please, sir?

**Comment by jamesshore** [comment]
Fixed, thank you.

**Comment by jamesshore** [comment]
Updated code to work with Node LTS (8.9.4). It will probably work with 9.4.0 now if you use the `loose=true` build option.

**Comment by AppreciateIT** [comment]
Thank you sir, I'll give it a shot. :-)

**Comment by AppreciateIT** [comment]
Hmm, tried again, and no joy.  Same error: 
sudo ./jake.sh karma loose=true --trace

jake aborted.
TypeError: Cannot read property 'prototype' of undefined
    at Object.<anonymous> (/PhpstormProjects/ObjectPlayground/object_playground/node_modules/karma/node_modules/socket.io/lib/store.js:35:42)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/PhpstormProjects/ObjectPlayground/object_playground/node_modules/karma/node_modules/socket.io/lib/manager.js:16:13)
    at Module._compile (module.js:660:30)

**Comment by jamesshore** [comment]
Sorry, can't help you.

**Comment by AppreciateIT** [comment]
Cool that they have a "confused" emoji.  haha
Thanks anyway James.  Must be something on my end.

**Comment by AppreciateIT** [comment]
The issue seems to be this.
https://github.comkljf/LearnBoost/websocket.io/issues/55

I fixed it in object_playground/node_modules/karma/node_modules/socket.io/lib/store.js:35:42
Now I get a new error, maybe the same in a different place. 
Let me see... 

```
Cannot read property 'prototype' of undefined
    at Object.<anonymous> (/PhpstormProjects/ObjectPlayground/object_playground/node_modules/karma/node_modules/socket.io/lib/transports/websocket/hybi-07-12.js:442:43
```
Yup line 13 of hybi-07-12.js
line 12 hybi-16.js
line 12 default.js
line 14 in lib/socket.js
line 6 in namespace.js
24 in manager.js 
and line 254 in node_modules/karma/node_modules/socket.io/node_modules/policyfile/lib/server.js:254:34

**Comment by jamesshore** [comment]
You probably just have outdated files. Run `git clean -fdx && git reset --hard && git pull && git checkout master` to reset your repository, then make sure you're running the correct version of Node, then run `./jake.sh karma`. It should all work fine.

**Comment by AppreciateIT** [comment]
...and we have liftoff!

`node node_modules/karma/bin/karma start ./build/config/karma.conf.js
INFO [karma]: Karma v0.12.31 server started at  @http://localhost:9876/`

@jamesshore Thank you!

**Comment by jamesshore** [comment]
Your files are still outdated. You're running Karma 0.12, which doesn't work on current versions of Node. When I fixed the issue above, I upgraded Karma to 2.0.0. Again, you need to run `git clean -fdx && git reset --hard && git pull && git checkout master` to reset your repository and get the current code, *which includes the correct version of Karma*, then `./jake.sh karma` should start Karma for you without issue.
